packages/slate/src/utils/memoize.js

```diff
@@ -108,11 +125,47 @@ function memoize(object, properties, options = {}) {
       if (takesArguments) {
         this.__cache = setIn(this.__cache, keys, v)
       } else {
-        this.__cache.set(property, v)
+        this.__cache_no_args[property] = v
       }
 
       return value
     }
+
+    // Keep fn.length unchanged after cache binding
```

Comment:
What is this stuff for?

Reply:
The problem of 
```
object["getAncestors"] === function(...args) 
```
is 

object["getAncestors"].length will be 0 regardless of the original function length.
So I have to hardcode that staff to reset the function length

Another trick here is about 
```
function getLeaves(decoration = [])
// getLeaves.length === 0
```
The argument with default params are not counted by function length.  Then I need to pass (...args) in the following in case there are functions with default param length.

Reply:
Gotcha. I think this sounds like a reason to not simplify the API then, and to keep the existing code as-is? Because we're dealing with a leaky abstraction.

Reply:
@ianstormtaylor The existing code does not keep the func.length either, because the existing `function (...args)` always have `object[property].length === 0`

I can remove this part if you think keeping func.length is not necessary.

Reply:
One advantage of removing takeArguments is that we can allow getLeaves use no arguments cache when with the default params. This is about 100% benchmark speed up in the benchmark test.

Reply:
Ah I see what you mean. I think we're good to ignore handling the function length here, since we weren't doing it before.

Reply:
@ianstormtaylor Fixed~